src/Synth/ADnote.cpp

```diff
@@ -448,14 +585,27 @@ void ADnote::setupVoiceMod(int nvoice, bool first_run)
                 || (voice.FMEnabled == FMTYPE::RING_MOD))
             tmp = getFMvoicebasefreq(nvoice);
 
-        if(!pars.GlobalPar.Hrandgrouping)
-            pars.VoicePar[vc].FmGn->newrandseed(prng());
+        if(!waveTables) // debug code for testing only
+            if(!pars.GlobalPar.Hrandgrouping)
+                pars.VoicePar[vc].FmGn->newrandseed(prng());
 
         for(int k = 0; k < voice.unison_size; ++k)
-            voice.oscposhiFM[k] = (voice.oscposhi[k]
-                    + pars.VoicePar[vc].FmGn->get(
-                        voice.FMSmp, tmp))
-                % synth.oscilsize;
+        {
+            if(waveTables)
+            {
```

Comment:
This seems to be doing redundant copies and the ->getFinalOutpos()::char data type issue seems to reoccur here. Does some of this need to be moved outside of the loop?

Reply:
First, I thought this was already bad design in the old code. If you would separate getting the "get" and the "out position calculation" in the old code, calling "get" once should totally suffice. But then, it seems that the old code calls `get` multiple times in order to get waves with different random for the different unison voices - and so the WaveTable should get waves based on different random, too. Does that make sense?

Reply:
Yep, it makes sense. It mostly surprises me because I had some wrong assumptions about unison. Thanks for correcting my misunderstanding with the code examples.